### DISCUSSION TITLE
Map GPIO to different pins

### DISCUSSION THREAD

**Discussion by gmag11** [Mair/esp32-course#122] [Course topic requests]
Hi, I am using a M5Stick-c plus module that only exposes pins 25, 26 and 36. I want to use touch sensors but these 3 pins are not in the ones that support that function.

I know that esp32 includes a IO MUX but I have not a cleas view on how to map functions to pins.

This seems to be possible with PIN_FUNC_SELECT macro. How could I do to map any touch sensor to pins that correspond to GPIO 25, 26 or 36?

Would you add something about this topic? Thanks!

**Comment by Mair** [comment]
hi @gmag11 
Unfortunately none of those pins (25,26 or 36) are touch capable :( 
If you look at the data sheet https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf on section 4.2 you will find this table
![image](https://user-images.githubusercontent.com/2477839/174546404-a0fdeb9f-312e-42d7-942c-84dc0eb0a1bc.png)
As you can see your pins or not in the list so are not connected.

Have a look at https://i0.wp.com/randomnerdtutorials.com/wp-content/uploads/2018/08/esp32-pinout-chip-ESP-WROOM-32.png?quality=100&strip=all&ssl=1 and focus on the purple

**Reply by gmag11** [reply]
As far as I've found only digital functions are routed through GPIO MUX. Analog like functions (like ADC or touch sensor) cannot be mapped to other pins. Thanks @Mair
